contributor-docs/release-guide.md

```diff
@@ -17,44 +14,49 @@ limitations under the License.
+The Apache Beam project periodically declares and publishes releases.  A
+release is one or more packages of the project artifact(s) that are approved
+for general public distribution and use.  They may come with various degrees of
+caveat regarding their perceived quality and potential for change, such as
+“alpha”, “beta”, “incubating”, “stable”, etc.
 
-The Beam community treats releases with great importance.
-They are a public face of the project and most users interact with the project only through the releases. Releases are signed off by the entire Beam community in a public vote.
+The Beam community treats releases with great importance.  They are a public
+face of the project and most users interact with the project only through the
+releases. Releases are signed off by the entire Beam community in a public
+vote.
 
-Each release is executed by a *Release Manager*, who is selected among the Beam committers.
-This document describes the process that the Release Manager follows to perform a release.
-Any changes to this process should be discussed and adopted on the [dev@ mailing list](/get-started/support/).
+Each release is executed by a *Release Manager*, who is selected among the Beam
+committers.  This document describes the process that the Release Manager
+follows to perform a release.  Any changes to this process should be discussed
+and adopted on the [dev@ mailing list](/get-started/support/).
 
-Please remember that publishing software has legal consequences.
-This guide complements the foundation-wide [Product Release Policy](https://www.apache.org/dev/release.html) and [Release Distribution Policy](https://www.apache.org/dev/release-distribution).
+Please remember that publishing software has legal consequences.  This guide
+complements the foundation-wide [Product Release
+Policy](https://www.apache.org/dev/release.html) and [Release Distribution
+Policy](https://www.apache.org/dev/release-distribution).
 
 ### Overview
 
-<img src="/images/release-guide-1.png" alt="Release step flow chart" width="100%">
+<img src="images/release-guide-1.png" alt="Release step flow chart" width="100%">
 
 The release process consists of several steps:
 
 1. Decide to release
-1. Prepare for the release
-1. Build a release candidate
-    1. Create a release branch
-    1. Verify release branch
-    1. Tag a release candidate
-    1. Build release artifacts
-1. Vote on the release candidate
-    * During vote process, run validation tests
-1. If necessary, fix any issues and go back to step 3.
-1. Finalize the release
-1. Promote the release
-
-**********
+2. Prepare for the release
+3. Build a release candidate
+   1. Create a release branch
+   2. Verify release branch
+   3. Tag a release candidate
+   4. Build release artifacts
+4. Vote on the release candidate
+   * During vote process, run validation tests
+5. If necessary, fix any issues and go back to step 3.
+6. Finalize the release
+7. Promote the release
```

Comment:
Nit: I don't really like this change; using `1.` for everything leads to the same result and makes it easier to insert or remove things later with a minimal diff

Reply:
Strong disagree. Using 1 for everything makes the file unreadable without a renderer, and relegates markdown to only a source language, whereas it was (rightly) invented to be a standard way of writing text that could optionally be rendered. My nit: the hack to just use `1.` should never have been done in any file on the whole internet :-)

Reply:
I'm with Kenn on this one. Minimizing diffs in this context is the wrong priority.

Reply:
Minimizing diffs is part of it, but also you're less likely to accidentally get things wrong.

> Using 1 for everything makes the file unreadable without a renderer

Is it really unreadable? If you're doing a lot of referencing like `after step #1 do X`, then maybe, but the intent of ordering is still there.


With all that said, I'm not super interested in getting into an opinion-driven argument here and will certainly not block on this.

Reply:
I will also at least point out that I consulted 4 different markdown style guides and all recommended numbering for long lists (which would include this one) if you don't have a formatting tool that takes care of fixing numbers automatically. Some recommended it for short lists too (e.g. https://cirosantilli.com/markdown-style-guide/#ordered). Again, that just means that others have my opinion (not that its correct) and the tradeoff here is somewhat obvious.

Reply:
I think the main axiom that is opinion-driven is whether _this_ markdown file is to be a text format or just source for a rendering. And given a population of people on both sides of that, which we obviously have, the next opinion-driven decision is whether to accommodate both or not.

As a text format, the question of "long enough to switch to lazy numbering is moot. A list of `1.` is not a numbered list. It just has a clunky bullet.

This list can just be made better instead:

 - it has a reference "go back to step 3" that would not refer to anything if there is no "3". TBH that could just be clearer with words, so I switched it.
 - it has an extraneous nested list which is also numbered, super gross and even rendered that way by our website (GitHub does better)
 - the outer and inner nested lists are both incorrect!

Reply:
Random aside:

The guideline for when to use a numbered list vs bullets in writing is whether or not the order of the items matters. But in the case of programmers applying their "source/render/diff" brain to markdown they've invented a third category which might be useful if it really came into being as a thing in writing: a kind of bullet that indicates that the order matters, but that order is to be the textual order which was there all along. I'll say that I think ↳ is probably a nice symbol that communicates that intent more elegantly.

So, yea, I have maybe thought too much about this, because I care about writing and reading and language. I think the actual thing that has been created it way more interesting than its creators intended :-)

Reply:
I'm going to call out that I did the bad thing of saying "This isn't important enough to argue about", and then presented multiple arguments for why we should do it my way, but I do genuinely think this is not worth the discussion when really we just seem to have different values here (and probably the next person to edit the file will just change it anyways since neither of us is really talking about something systemic here).

With that said, I am going to bow out of this conversation instead of continuing because I don't want to stand in the way of you making things better with your PR.